centralstore/local_store.go

```diff
@@ -391,8 +408,6 @@ func (lrs *LocalStore) KeepTraces(ctx context.Context, statuses []*CentralTraceS
 			// use the one we have in the statuses list
 			lrs.states[DecisionKeep][status.TraceID] = status
 			delete(lrs.states[AwaitingDecision], status.TraceID)
-			// also remove it from the current traces list
```

Comment:
Why do we want to keep the span list after a decision has been made?

Reply:
I moved this to the cleanup stage because we need it for late spans.